habitat-lab/habitat/tasks/rearrange/actions/actions.py

```diff
@@ -386,6 +386,290 @@ def step(self, *args, is_last_action, **kwargs):
+        self.cur_robot.sim_obj.joint_velocities = set_dat["vel"]
+        self.cur_robot.sim_obj.joint_forces = set_dat["pos"]
+
+    def update_base(self):
+        """
+        Update the robot base
+        """
+
+        ctrl_freq = self._sim.ctrl_freq
+
+        before_trans_state = self._capture_robot_state()
+
+        trans = self.cur_robot.sim_obj.transformation
+        rigid_state = habitat_sim.RigidState(
+            mn.Quaternion.from_matrix(trans.rotation()), trans.translation
+        )
+
+        target_rigid_state = self.base_vel_ctrl.integrate_transform(
+            1 / ctrl_freq, rigid_state
+        )
+        end_pos = self._sim.step_filter(
+            rigid_state.translation, target_rigid_state.translation
+        )
+
+        target_trans = mn.Matrix4.from_(
+            target_rigid_state.rotation.to_matrix(), end_pos
+        )
+        self.cur_robot.sim_obj.transformation = target_trans
+
+        if not self._config.get("allow_dyn_slide", True):
+            # Check if in the new robot state the arm collides with anything.
+            # If so we have to revert back to the previous transform
+            self._sim.internal_step(-1)
+            colls = self._sim.get_collisions()
+            did_coll, _ = rearrange_collision(
+                colls, self._sim.snapped_obj_id, False
+            )
+            if did_coll:
+                # Don't allow the step, revert back.
+                self._set_robot_state(before_trans_state)
+                self.cur_robot.sim_obj.transformation = trans
+        if self.cur_grasp_mgr.snap_idx is not None:
+            # Holding onto an object, also kinematically update the object.
+            # object.
+            self.cur_grasp_mgr.update_object_to_grasp()
+
+    def step(self, *args, is_last_action, **kwargs):
+        lin_vel, ang_vel = kwargs[self._action_arg_prefix + "base_vel"]
+        lin_vel = np.clip(lin_vel, -1, 1) * self._config.lin_speed
+        ang_vel = np.clip(ang_vel, -1, 1) * self._config.ang_speed
+        if not self._config.allow_back:
+            lin_vel = np.maximum(lin_vel, 0)
+
+        trans = self._sim.robot.base_transformation
+        global_pos = trans.transform_point(np.array([lin_vel, 0, 0]))
+
+        # Set the goal
+        base_pose_goal = [
+            global_pos[0],
+            global_pos[2],
+            float(self._sim.robot.base_rot)+ang_vel,
+        ]
+        self.controller.set_goal(base_pose_goal)
+
+        # Get the current position
+        base_pose = [
+            self._sim.robot.base_pos[0],
+            self._sim.robot.base_pos[2],
+            float(self._sim.robot.base_rot),
+        ]
+        base_action = self.controller.forward(base_pose, self._sim)
+
+        self.base_vel_ctrl.linear_velocity = mn.Vector3(base_action[0], 0, 0)
+        self.base_vel_ctrl.angular_velocity = mn.Vector3(0, base_action[1], 0)
+
+        if lin_vel != 0.0 or ang_vel != 0.0:
```

Comment:
should this be something with a tolerance?

Reply:
I will fix this, thanks!